idaes/models/properties/modular_properties/base/generic_property.py

```diff
@@ -1653,10 +1653,14 @@ def fix_initialization_states(self):
 
         # Also need to deactivate sum of mole fraction constraint
         for k in self.values():
-            try:
-                k.sum_mole_frac_out.deactivate()
-            except AttributeError:
-                pass
+            if not k.config.defined_state:
+                try:
+                    k.sum_mole_frac_out.deactivate()
+                except AttributeError:
+                    try:
```

Comment:
I wonder if it would be useful to add logger messages here for the user's awareness. For example, "Deactivating sum_mole_frac_out constraint" followed by "sum_mole_frac_out constraint does not exist. Deactivating equilbrium_constraint."

Reply:
Is this constraint activated after the initialization?